### PR TITLE
feat(postgres): Backport #9883 to v4

### PR DESCRIPTION
**Pull request by jhiner**
### Pull Request check-list

_Please make sure to review and check all of these items:_

- [ ] Does `npm run test` or `npm run test-DIALECT` pass with this change (including linting)?
- [x] Does the description below contain a link to an existing issue (Closes #[issue]) or a description of the issue you are solving?
- [ ] Have you added new tests to prevent regressions? (N/A)
- [ ] Is a documentation update included (if this change modifies existing APIs, or introduces new ones)? (N/A)
- [x] Did you follow the commit message conventions explained in [CONTRIBUTING.md](https://github.com/sequelize/sequelize/blob/master/CONTRIBUTING.md)?

### Description of change

Backporting #9883 to v4. Cherry-picked 3a56cf74 with conflicts resolved.

### CODE REVIEW

**Comment by jhiner** [comment]
I am not sure why the build is failing 🤔  Any tips are appreciated. Thanks

**Comment by sushantdhiman** [comment]
You can check error logs from CI, currently it fails for `[].includes` which is not available in Node v4

File: lib/dialects/postgres/index.js [hunk 1]
```diff
@@ -40,6 +40,9 @@ PostgresDialect.prototype.supports = _.merge(_.cloneDeep(AbstractDialect.prototy
     using: 2,
     where: true
   },
+  inserts: {
```
Comment (sushantdhiman) [inline lib/dialects/postgres/index.js]: `inserts` should be removed from here as well, or postgres wont override correct option

File: lib/model.js [hunk 2]
```diff
@@ -2310,10 +2310,10 @@ class Model {
...
-   * @param  {Boolean}      [options.hooks=true]             Run before / after bulk create hooks?
-   * @param  {Boolean}      [options.individualHooks=false]  Run before / after create hooks for each individual Instance? BulkCreate hooks will still be run if options.hooks is true.
-   * @param  {Boolean}      [options.ignoreDuplicates=false] Ignore duplicate values for primary keys? (not supported by postgres)
+   * @param  {boolean}      [options.validate=false]         Should each row be subject to validation before it is inserted. The whole insert will fail if one row fails validation
+   * @param  {boolean}      [options.hooks=true]             Run before / after bulk create hooks?
+   * @param  {boolean}      [options.individualHooks=false]  Run before / after create hooks for each individual Instance? BulkCreate hooks will still be run if options.hooks is true.
+   * @param  {boolean}      [options.ignoreDuplicates=false] Ignore duplicate values for primary keys? (not supported by postgres < 9.5)
```
Comment (sushantdhiman) [inline lib/model.js]: Remove `Boolean` casing change, just include message change part i.e. `< 9.5`

**Comment by sushantdhiman** [comment]
> /home/travis/build/sequelize/sequelize/lib/dialects/abstract/index.js
  44:1  error  Expected indentation of 2 spaces but found 3  indent

Run `npm run lint` and fix linting issue

**Comment by jhiner** [comment]
Thanks so much!

File: test/unit/dialects/postgres/query-generator.test.js [hunk 3]
```diff
@@ -557,9 +557,21 @@ if (dialect.match(/^postgres/)) {
         {
           arguments: ['myTable', {name: 'foo'}],
           expectation: "INSERT INTO \"myTable\" (\"name\") VALUES ('foo');"
+        }, {
+          arguments: ['myTable', {name: 'foo'}, {}, { ignoreDuplicates: true }],
+          expectation: {
+            query: 'INSERT INTO "myTable" ("name") VALUES ($1) ON CONFLICT DO NOTHING;',
+            bind: ['foo']
```
Comment (sushantdhiman) [inline test/unit/dialects/postgres/query-generator.test.js]: v4 does not support automatic `bind` with create. You need to rewrite this something like this for v4

```js
arguments: ['myTable', {name: 'foo'}, {}, { ignoreDuplicates: true }],
expectation: 'INSERT INTO "myTable" ("name") VALUES ('foo') ON CONFLICT DO NOTHING;'
```

**Pull request merged**